### PR TITLE
Jonathan/test attestation record

### PR DESCRIPTION
**Pull request by JonathanLorimer**
Hey Reviewer,

This is my first PR on your project and any other open source project for that matter, so please have patience. Hopefully your time will be repaid by my productivity in the future.

A few points:
- I read the contributing.md file but couldn't find a style guide so hopefully the way I have written the javascript works for you
- Originally I only wanted to write tests, but the attestationRecord had a couple syntax errors I ended up having to fix, so hopefully this PR isn't too large
- I don't really understand the underlying technology (I am reading the sharding and Casper papers currently) so I just wrote tests to verify the class constructor's functionality; no imaginative test cases

PR is in response to this issue: https://github.com/ChainSafeSystems/lodestar_chain/issues/12

Thank you for taking the time,
Jonathan

### CODE REVIEW

File: core/tests/test_attestationRecord.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+const missingFields = {
+    slot: 4,
+    oblique_parent_hashes: [new Buffer(32)],
+    attester_bitfield: new Buffer(''),
+    justified_block_hash: new Buffer(32)
+}
+
+// Recursively compares items and handles for Arrays, Buffers, and Objects
+const deepCompare = (a, b) => {
```
Comment (ansermino) [inline core/tests/test_attestationRecord.js]: Does this differ from chai.assert.deepCopy?

**Review by ansermino — COMMENTED** [review]
Hey @JonathanLorimer :) Thanks so much for your contribution! 

The first thing to mention is we are in the process of migrating to Typescript (#39). Unfortunately this may render some of your work unusable but will hopefully lead to great benefits for the project in the future. I don't see an TS experience on your profile but please do not let this be a deterrent, many of us are learning as we go.

To address your points:
- True! Typescript will definitely mitigate some of this but establishing some conventions for things such as comments, PR's,  and file layouts would be great. Please (if you want) open an issue and share your thoughts.
- Indeeed! Should see some fixes in #39 and I believe the spec itself has changed and those changes will be reflected.
- The functionality is still very basic, tests for simple things are great! 

Lets wait for #39 to be closed and then we can reassess this. Thank you again for your work and hope to see more in the future :+1:

File: core/tests/test_attestationRecord.js (hunk 1 above)
Comment (JonathanLorimer) [reply core/tests/test_attestationRecord.js]: I hadn't seen that before, but you are right, chai.assert.deepEqual does the trick

**Comment by GregTheGreek** [comment]
@JonathanLorimer Thanks for contributing! I back all of @ansermino points.

@ansermino on that note perhaps for #39 we should delete the old code?

**Pull request closed without merging**